purchases/src/main/kotlin/com/revenuecat/purchases/google/usecase/QueryProductDetailsUseCase.kt

```diff
@@ -114,4 +119,37 @@ internal class QueryProductDetailsUseCase(
             responseTime = Duration.between(requestStartTime, useCaseParams.dateProvider.now),
         )
     }
+
+    // This error was introduced in Google's Billing Client 7.1.0 and 7.1.1. Logging an error to notify the developer,
+    // as soon as possible that purchases will fail unless core library desugaring is enabled.
+    private companion object {
+        val hasLoggedBillingFlowParamsError = AtomicBoolean(false)
+    }
+
+    @Suppress("NestedBlockDepth")
+    private fun logErrorIfIssueBuildingBillingParams(productDetails: List<ProductDetails>) {
+        productDetails.firstOrNull()?.let {
+            if (hasLoggedBillingFlowParamsError.getAndSet(true)) {
+                return
+            }
+            val productDetailsParams = try {
+                val offerToken = it.subscriptionOfferDetails?.firstOrNull()?.offerToken
+                ProductDetailsParams.newBuilder()
+                    .setProductDetails(it)
+                    .apply { if (offerToken != null) setOfferToken(offerToken) }
+                    .build()
+            } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
```

Comment:
Added this just in case there is something I've missed... I've tested this with subscriptions and one time products, but I don't like how this has a bunch of validations internally that we need to acomodate or crash... This catch is to make sure we don't accidentally break.